readthedocs/proxito/views/serve.py

```diff
@@ -89,6 +89,24 @@ def get(self,
         ``subproject_slash`` is used to determine if the subproject URL has a slash,
         so that we can decide if we need to serve docs or add a /.
         """
+        unresolved_domain = request.unresolved_domain
+        # Handle requests that need canonicalizing,
+        # e.g. HTTP -> HTTPS, redirect to canonical domain, etc.
+        redirect_type = self._get_canonical_redirect_type(request)
+        if redirect_type:
+            # Attach the current project to the request.
+            request.path_project_slug = unresolved_domain.project.slug
+            try:
+                return self.canonical_redirect(
+                    request=request,
+                    final_project=unresolved_domain.project,
+                    external_version_slug=unresolved_domain.external_version_slug,
+                    redirect_type=redirect_type,
+                )
+            except InfiniteRedirectException:
+                # Don't redirect in this case, since it would break things.
```

Comment:
Probably want to log a warning here? Blank exception `pass`'s scare me..

Reply:
We already log in the method itself https://github.com/readthedocs/readthedocs.org/blob/1ab17590966f42a9efcafbe17f800b90fdf78ec7/readthedocs/proxito/views/mixins.py#L385-L391

Reply:
I also mentioned this in other PRs.

In the way the logic is implemented now, we are raising exception from the unresolver for cases we are going to handle. So, the unresolver _always_ logs the exception and the method handling it, just passes.

This is what I understood from the reviews I've done. Correct me @stsewd if I'm wrong.

Reply:
Why are we raising the exception and then ignoring it? The comment can definitely be updated to explain this architecture... something like:

```suggestion
                # ``canonical_redirect`` raises this when it's redirecting back to itself.
                # We can safely ignore it here because it's logged in ``canonical_redirect``.
```